bin/doctest

```diff
@@ -2,18 +2,19 @@
 
 'use strict';
 
-// Based on:
-// https://github.com/jashkenas/coffeescript/blob/1.9.1/src/command.coffee#L431-L441
-var args = process.argv.slice (1);
+var path = require ('path');
+
+var args = process.argv.slice (2);
 var idx = args.indexOf ('--nodejs');
-if (idx < 0 || idx === args.length - 1) {
-  require ('../lib/command');
-} else {
-  require ('child_process')
-  .spawn (process.execPath,
-          args[idx + 1]
-            .split (/\s+/)
-            .concat (args.slice (0, idx), args.slice (idx + 2)),
-          {cwd: process.cwd (), env: process.env, stdio: [0, 1, 2]})
-  .on ('exit', process.exit);
-}
+var esm = Number (process.versions.node.split ('.')[0]) >= 9;
+var flags = idx >= 0 && idx < args.length - 1;
+
+require ('child_process').spawn (
+  process.execPath,
+  [].concat (esm ? ['--experimental-modules'] : [])
+    .concat (flags ? args[idx + 1].split (/\s+/) : [])
+    .concat ([path.resolve (__dirname, '..', 'lib', 'command')])
```

Comment:
If my understanding of this code is correct, it seems advisable to `.concat (['--'])` before appending the file path to prevent the file path from being interpreted as an option name.

Reply:
We could. I just left the behaviour as close to the original as possible.

Reply:
Okay. I'll make this change in a separate pull request. :)

Reply:
:zap:

Reply:
I saw your comment too late. Want me to revert?

Reply:
I'm happy for this change to be part of the pull request, as it's trivial and tangentially related to your changes.